src/layers/transform/pooling.cpp

```diff
@@ -114,15 +114,16 @@ void pooling_layer<TensorDataType, Layout, Device>::fp_compute_dnn()
   LBANN_ERROR("DNN library not detected");
 #else
   using ScalingType = dnn_lib::ScalingParamType<TensorDataType>;
+  using MatrixType = El::Matrix<TensorDataType, El::Device::GPU>;
   const auto& local_input = this->get_local_prev_activations();
   auto& local_output = this->get_local_activations();
   if (local_input.Height() > 0 && local_input.Width() > 0) {
     // Initialize GPU workspace
-    El::Matrix<TensorDataType, El::Device::GPU> workspace;
     size_t workspace_size =
       dnn_lib::get_pooling_ws_size(m_pooling_dnn_desc,
-                                  m_tensors_dnn_desc.get_activations());
-    workspace.Resize(workspace_size / sizeof(TensorDataType), 1);
+                                   m_tensors_dnn_desc.get_activations());
+    m_workspace.reset(new MatrixType());
```

Comment:
I don't like this behavior (I'd rather keep a workspace matrix on storage mode 1 and Empty/Resize it), but this is what used to exist before.